client/llb/definition_test.go

```diff
@@ -0,0 +1,53 @@
+package llb
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestDefinitionEquivalence(t *testing.T) {
+	for _, tc := range []struct {
+		name  string
+		state State
+	}{
+		{"scratch", Scratch()},
+		{"image op", Image("ref")},
+		{"exec op", Image("ref").Run(Shlex("args")).Root()},
+		{"local op", Local("name")},
+		{"git op", Git("remote", "ref")},
+		{"http op", HTTP("url")},
+		{"file op", Scratch().File(Mkdir("foo", 0600).Mkfile("foo/bar", 0600, []byte("data")).Copy(Scratch(), "src", "dst"))},
```

Comment:
Could you add a test that sets a platform to the original state and then checks it after `NewState(defOp)`

Reply:
@tonistiigi Can you clarify what you mean by `original state`?

Reply:
The state that is marshaled and then passed to `NewDefinitionOp`

Reply:
This needs a specific call to `state.Platform()` to verify as the platform is stored in the state directly, not only in the proto where digest comparison is enough.